starknet/src/types/indexed_strategy.cairo

```diff
@@ -38,3 +38,39 @@ impl IndexedStrategyImpl of IndexedStrategyTrait {
         };
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::{IndexedStrategy, IndexedStrategyTrait};
+
+    #[test]
+    #[available_gas(100000000)]
+    fn no_duplicates() {
+        array![
+            IndexedStrategy { index: 0_u8, params: array![1, 2, 3, 4], },
+            IndexedStrategy { index: 1_u8, params: array![1, 2, 3, 4], },
+            IndexedStrategy { index: 2_u8, params: array![1, 2, 3, 4], },
+        ]
+            .span()
+            .assert_no_duplicate_indices();
+    }
+
+    #[test]
+    #[available_gas(100000000)]
+    fn empty_array() {
+        array![].span().assert_no_duplicate_indices();
+    }
+
+    #[test]
+    #[available_gas(100000000)]
+    #[should_panic(expected: ('Duplicate Found',))]
+    fn catch_duplicates() {
```

Comment:
for the sake of completeness we might want to add a test where the duplicates are the first and last (256th) elements

Reply:
added https://github.com/snapshot-labs/sx-starknet/pull/577/commits/2f674d84a7e87907bb58d156fbe16e30310584db